Refactoring: extract EmbedThisPanel class from CommonViewer

ndushay opened the pull request:
EmbedThisPanel class handles the embed panel markup.   It is modeled on the DownloadPanel class.

Closes #544

ndushay commented on lib/embed/embed_this_panel.rb:
```diff
@@ -0,0 +1,58 @@
+module Embed
+  class EmbedThisPanel
+    def initialize(druid:, height:, width:, purl_object_title:, &block)
```
I couldn't think of a better way to get all this information to this class.  5 args, and 5 instance variable feels like sort of a lot.

jkeck commented:
@ndushay this needs a rebase after merging the rubocop fixes.

ndushay commented:
rebased.

jkeck commented on lib/embed/embed_this_panel.rb:
```diff
@@ -0,0 +1,58 @@
+                  </div>
+                  <div class='sul-embed-section'>
+                    <textarea id='sul-embed-iframe-code' data-behavior='iframe-code' rows=4>
+                      <iframe src='#{Settings.embed_iframe_url}?url=#{Settings.purl_url}/#{druid}' height='#{height}px' width='#{width || height}px' frameborder='0' marginwidth='0' marginheight='0' scrolling='no' allowfullscreen></iframe>
```
Unfortunately it looks like the heredoc spacing in the `<textarea>` is making its way into the actual form.

<img width="521" alt="textarea" src="https://cloud.githubusercontent.com/assets/96776/14578174/f758cbbe-0335-11e6-953e-a93728497466.png">

jkeck commented on lib/embed/embed_this_panel.rb:
```diff
@@ -0,0 +1,58 @@
+              <div class='sul-embed-embed-this-form'>
+                <span class='sul-embed-options-label'>Select options:</span>
+                <div class='sul-embed-section sul-embed-embed-title-section'>
+                  <input type='checkbox' id='sul-embed-embed-title' 'data-embed-attr'='hide_title' checked=true />
```
The `data-embed-attr` should not have the single quotes around it.  This is currently causing the title option to add the parameter `undefined=true` instead of `hide_title=true` (there should have been a test for this, looks like that one got away from us).

<img width="542" alt="undefined-title" src="https://cloud.githubusercontent.com/assets/96776/14578189/42ae2d34-0336-11e6-833e-fd226dccb511.png">

ndushay replied:
ok, will fix.

jkeck replied:
More than happy to to take a look at this with you on Monday!

jkeck commented:
@ndushay noticed a few things.  Some inline comments, but here are two screenshots that illustrate the issues.

### Expected
<img width="529" alt="expected-embed" src="https://cloud.githubusercontent.com/assets/96776/14578197/76969ff0-0336-11e6-8cd4-54af2388f780.png">

### Actual
<img width="542" alt="undefined-title" src="https://cloud.githubusercontent.com/assets/96776/14578198/7ee83f74-0336-11e6-96d2-d8925a6ab5a1.png">

ndushay commented:
@jkeck   I figured out how to see what you were looking at in my browser.  I made the corrections you indicated, and added a test for the hide_title checkbox.   I also rebased for good measure.   Let me know if you have any other suggestions.   I still haven't come up with a way to avoid 5 args and 5 instance vars, but since I think this PR is an improvement on the existing code base, maybe that's a problem for another PR.

ndushay commented:
tweaked it a bit more per F2F with @jkeck.
